extensions/ql-vscode/src/model-editor/modeled-method.ts

```diff
@@ -109,3 +110,19 @@ export function modeledMethodSupportsProvenance(
     modeledMethod.type === "neutral"
   );
 }
+
+export function isModelAccepted(
+  modeledMethod: ModeledMethod | undefined,
+  modelingStatus: ModelingStatus,
+): boolean {
+  if (!modeledMethod) {
+    return true;
+  }
+
+  return (
+    modelingStatus !== "unsaved" ||
+    modeledMethod.type === "none" ||
```

Comment:
I think this check is now redundant because `none` modeled methods do not support provenance, but leaving this explicit in might be good to make this intention clear.

Reply:
Yeah, agreed!